<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <title>Test Report</title>
    <link href="assets/style.css" rel="stylesheet" type="text/css"/></head>
  <body onLoad="init()">
    <script>/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */


function toArray(iter) {
    if (iter === null) {
        return null;
    }
    return Array.prototype.slice.call(iter);
}

function find(selector, elem) {
    if (!elem) {
        elem = document;
    }
    return elem.querySelector(selector);
}

function find_all(selector, elem) {
    if (!elem) {
        elem = document;
    }
    return toArray(elem.querySelectorAll(selector));
}

function sort_column(elem) {
    toggle_sort_states(elem);
    var colIndex = toArray(elem.parentNode.childNodes).indexOf(elem);
    var key;
    if (elem.classList.contains('numeric')) {
        key = key_num;
    } else if (elem.classList.contains('result')) {
        key = key_result;
    } else {
        key = key_alpha;
    }
    sort_table(elem, key(colIndex));
}

function show_all_extras() {
    find_all('.col-result').forEach(show_extras);
}

function hide_all_extras() {
    find_all('.col-result').forEach(hide_extras);
}

function show_extras(colresult_elem) {
    var extras = colresult_elem.parentNode.nextElementSibling;
    var expandcollapse = colresult_elem.firstElementChild;
    extras.classList.remove("collapsed");
    expandcollapse.classList.remove("expander");
    expandcollapse.classList.add("collapser");
}

function hide_extras(colresult_elem) {
    var extras = colresult_elem.parentNode.nextElementSibling;
    var expandcollapse = colresult_elem.firstElementChild;
    extras.classList.add("collapsed");
    expandcollapse.classList.remove("collapser");
    expandcollapse.classList.add("expander");
}

function show_filters() {
    var filter_items = document.getElementsByClassName('filter');
    for (var i = 0; i < filter_items.length; i++)
        filter_items[i].hidden = false;
}

function add_collapse() {
    // Add links for show/hide all
    var resulttable = find('table#results-table');
    var showhideall = document.createElement("p");
    showhideall.innerHTML = '<a href="javascript:show_all_extras()">Show all details</a> / ' +
                            '<a href="javascript:hide_all_extras()">Hide all details</a>';
    resulttable.parentElement.insertBefore(showhideall, resulttable);

    // Add show/hide link to each result
    find_all('.col-result').forEach(function(elem) {
        var extras = elem.parentNode.nextElementSibling;
        var expandcollapse = document.createElement("span");
        if (elem.innerHTML === 'Passed') {
            extras.classList.add("collapsed");
            expandcollapse.classList.add("expander");
        } else {
            expandcollapse.classList.add("collapser");
        }
        elem.appendChild(expandcollapse);

        elem.addEventListener("click", function(event) {
            if (event.currentTarget.parentNode.nextElementSibling.classList.contains("collapsed")) {
                show_extras(event.currentTarget);
            } else {
                hide_extras(event.currentTarget);
            }
        });
    })
}

function init () {
    reset_sort_headers();

    add_collapse();

    show_filters();

    toggle_sort_states(find('.initial-sort'));

    find_all('.sortable').forEach(function(elem) {
        elem.addEventListener("click",
                              function(event) {
                                  sort_column(elem);
                              }, false)
    });

};

function sort_table(clicked, key_func) {
    var rows = find_all('.results-table-row');
    var reversed = !clicked.classList.contains('asc');
    var sorted_rows = sort(rows, key_func, reversed);
    /* Whole table is removed here because browsers acts much slower
     * when appending existing elements.
     */
    var thead = document.getElementById("results-table-head");
    document.getElementById('results-table').remove();
    var parent = document.createElement("table");
    parent.id = "results-table";
    parent.appendChild(thead);
    sorted_rows.forEach(function(elem) {
        parent.appendChild(elem);
    });
    document.getElementsByTagName("BODY")[0].appendChild(parent);
}

function sort(items, key_func, reversed) {
    var sort_array = items.map(function(item, i) {
        return [key_func(item), i];
    });
    var multiplier = reversed ? -1 : 1;

    sort_array.sort(function(a, b) {
        var key_a = a[0];
        var key_b = b[0];
        return multiplier * (key_a >= key_b ? 1 : -1);
    });

    return sort_array.map(function(item) {
        var index = item[1];
        return items[index];
    });
}

function key_alpha(col_index) {
    return function(elem) {
        return elem.childNodes[1].childNodes[col_index].firstChild.data.toLowerCase();
    };
}

function key_num(col_index) {
    return function(elem) {
        return parseFloat(elem.childNodes[1].childNodes[col_index].firstChild.data);
    };
}

function key_result(col_index) {
    return function(elem) {
        var strings = ['Error', 'Failed', 'Rerun', 'XFailed', 'XPassed',
                       'Skipped', 'Passed'];
        return strings.indexOf(elem.childNodes[1].childNodes[col_index].firstChild.data);
    };
}

function reset_sort_headers() {
    find_all('.sort-icon').forEach(function(elem) {
        elem.parentNode.removeChild(elem);
    });
    find_all('.sortable').forEach(function(elem) {
        var icon = document.createElement("div");
        icon.className = "sort-icon";
        icon.textContent = "vvv";
        elem.insertBefore(icon, elem.firstChild);
        elem.classList.remove("desc", "active");
        elem.classList.add("asc", "inactive");
    });
}

function toggle_sort_states(elem) {
    //if active, toggle between asc and desc
    if (elem.classList.contains('active')) {
        elem.classList.toggle('asc');
        elem.classList.toggle('desc');
    }

    //if inactive, reset all other functions and add ascending active
    if (elem.classList.contains('inactive')) {
        reset_sort_headers();
        elem.classList.remove('inactive');
        elem.classList.add('active');
    }
}

function is_all_rows_hidden(value) {
  return value.hidden == false;
}

function filter_table(elem) {
    var outcome_att = "data-test-result";
    var outcome = elem.getAttribute(outcome_att);
    class_outcome = outcome + " results-table-row";
    var outcome_rows = document.getElementsByClassName(class_outcome);

    for(var i = 0; i < outcome_rows.length; i++){
        outcome_rows[i].hidden = !elem.checked;
    }

    var rows = find_all('.results-table-row').filter(is_all_rows_hidden);
    var all_rows_hidden = rows.length == 0 ? true : false;
    var not_found_message = document.getElementById("not-found-message");
    not_found_message.hidden = !all_rows_hidden;
}
</script>
    <p>Report generated on 11-Sep-2017 at 10:42:20 by<a href="https://pypi.python.org/pypi/pytest-html"> pytest-html</a> v1.15.2</p>
    <h2>Environment</h2>
    <table id="environment">
      <tr>
        <td>Packages</td>
        <td>{&apos;pytest&apos;: &apos;3.2.1&apos;, &apos;py&apos;: &apos;1.4.34&apos;, &apos;pluggy&apos;: &apos;0.4.0&apos;}</td></tr>
      <tr>
        <td>Platform</td>
        <td>Darwin-16.5.0-x86_64-i386-64bit</td></tr>
      <tr>
        <td>Plugins</td>
        <td>{&apos;metadata&apos;: &apos;1.5.0&apos;, &apos;json&apos;: &apos;0.4.0&apos;, &apos;html&apos;: &apos;1.15.2&apos;}</td></tr>
      <tr>
        <td>Python</td>
        <td>3.6.1</td></tr></table>
    <h2>Summary</h2>
    <p>31 tests ran in 7.28 seconds. </p>
    <p class="filter" hidden="true">(Un)check the boxes to filter the results.</p><input checked="true" class="filter" data-test-result="passed" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="passed">22 passed</span>, <input checked="true" class="filter" data-test-result="skipped" disabled="true" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="skipped">0 skipped</span>, <input checked="true" class="filter" data-test-result="failed" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="failed">9 failed</span>, <input checked="true" class="filter" data-test-result="error" disabled="true" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="error">0 errors</span>, <input checked="true" class="filter" data-test-result="xfailed" disabled="true" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="xfailed">0 expected failures</span>, <input checked="true" class="filter" data-test-result="xpassed" disabled="true" hidden="true" name="filter_checkbox" onChange="filter_table(this)" type="checkbox"/><span class="xpassed">0 unexpected passes</span>
    <h2>Results</h2>
    <table id="results-table">
      <thead id="results-table-head">
        <tr>
          <th class="sortable result initial-sort" col="result">Result</th>
          <th class="sortable" col="name">Test</th>
          <th class="sortable numeric" col="duration">Duration</th>
          <th>Links</th></tr>
        <tr hidden="true" id="not-found-message">
          <th colspan="4">No results found. Try to check the filters</th></tr></thead>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestFullConnection::()::test_connect_full1</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104cc6438&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104cc6240&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104cc6240&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestFullConnection object at 0x104cc69e8&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104cc6438&gt;<br/><br/>    def test_connect_full1(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Checks that all welcome messages are sent after the NICK and USER<br/>            commands are received.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client()<br/>    <br/>        client.send_cmd(&quot;NICK user1&quot;)<br/>        client.send_cmd(&quot;USER user1 * * :User One&quot;)<br/>    <br/>&gt;       irc_session.verify_welcome_messages(client, &quot;user1&quot;)<br/><br/>tests/chirc/tests/test_connection.py:397: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:444: in verify_welcome_messages<br/>    reply = self.get_reply(client, expect_code = replies.RPL_YOURHOST, expect_nick = nick, expect_nparams = 1)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104cc6438&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104cc6240&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestFullConnection::()::test_connect_full2</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104cce940&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104baa940&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104baa940&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestFullConnection object at 0x104b50d30&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104cce940&gt;<br/><br/>    def test_connect_full2(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Checks that all welcome messages are sent after the USER and NICK<br/>            commands are received.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client()<br/>    <br/>        client.send_cmd(&quot;USER user1 * * :User One&quot;)<br/>        client.send_cmd(&quot;NICK user1&quot;)<br/>    <br/>&gt;       irc_session.verify_welcome_messages(client, &quot;user1&quot;)<br/><br/>tests/chirc/tests/test_connection.py:410: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:444: in verify_welcome_messages<br/>    reply = self.get_reply(client, expect_code = replies.RPL_YOURHOST, expect_nick = nick, expect_nparams = 1)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104cce940&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104baa940&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestFullConnection::()::test_connect_full3</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104ccecc0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104cce6d8&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104cce6d8&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestFullConnection object at 0x104ccef28&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104ccecc0&gt;<br/><br/>    def test_connect_full3(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Checks that all welcome messages, as well as the LUSERS and<br/>            MOTD replies, are sent after the NICK and USER commands are received.<br/>            &quot;&quot;&quot;<br/>    <br/>        client = irc_session.get_client()<br/>    <br/>        client.send_cmd(&quot;NICK user1&quot;)<br/>        client.send_cmd(&quot;USER user1 * * :User One&quot;)<br/>    <br/>&gt;       irc_session.verify_welcome_messages(client, &quot;user1&quot;)<br/><br/>tests/chirc/tests/test_connection.py:423: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:444: in verify_welcome_messages<br/>    reply = self.get_reply(client, expect_code = replies.RPL_YOURHOST, expect_nick = nick, expect_nparams = 1)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104ccecc0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104cce6d8&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestMultiuserConnection::()::test_connect_2users</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104efb2e8&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104efb668&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104efb668&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestMultiuserConnection object at 0x104efb048&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104efb2e8&gt;<br/><br/>    def test_connect_2users(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects two clients to the server.<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:436: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:444: in verify_welcome_messages<br/>    reply = self.get_reply(client, expect_code = replies.RPL_YOURHOST, expect_nick = nick, expect_nparams = 1)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104efb2e8&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104efb668&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestMultiuserConnection::()::test_connect_duplicate_nick</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104d6d470&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104d6d978&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104d6d978&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestMultiuserConnection object at 0x104d6d278&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104d6d470&gt;<br/><br/>    def test_connect_duplicate_nick(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects two clients to the server, but the second<br/>            client tries to use the first client&#x27;s nickname<br/>            (and should get an ERR_NICKNAMEINUSE)<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       client1 = irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:446: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:444: in verify_welcome_messages<br/>    reply = self.get_reply(client, expect_code = replies.RPL_YOURHOST, expect_nick = nick, expect_nparams = 1)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104d6d470&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104d6d978&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestQUIT::()::test_quit_after_registration1</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104d28cc0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104d28f98&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104d28f98&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestQUIT object at 0x104edac88&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104d28cc0&gt;<br/><br/>    def test_quit_after_registration1(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects a client, and quits right after connecting.<br/>            Verifies the ERROR response, but not whether the server<br/>            actually disconnected the client.<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       client1 = irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:465: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:444: in verify_welcome_messages<br/>    reply = self.get_reply(client, expect_code = replies.RPL_YOURHOST, expect_nick = nick, expect_nparams = 1)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104d28cc0&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104d28f98&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestQUIT::()::test_quit_after_registration2</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104eded30&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104ede940&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104ede940&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestQUIT object at 0x104edea20&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104eded30&gt;<br/><br/>    def test_quit_after_registration2(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects a client, and quits right after connecting.<br/>            Verifies the ERROR response and also checks whether the<br/>            server disconnected the client.<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       client1 = irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:479: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:444: in verify_welcome_messages<br/>    reply = self.get_reply(client, expect_code = replies.RPL_YOURHOST, expect_nick = nick, expect_nparams = 1)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104eded30&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104ede940&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestQUIT::()::test_quit_after_registration3</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104b3b0b8&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104eb0f28&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104eb0f28&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestQUIT object at 0x104eb0c88&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104b3b0b8&gt;<br/><br/>    def test_quit_after_registration3(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects a client, and quits right after connecting.<br/>            QUIT does not specify a message, so the default &quot;Client Quit&quot; should<br/>            be assumed. Also verifies the ERROR response and also checks whether the<br/>            server disconnected the client.<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       client1 = irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:496: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:444: in verify_welcome_messages<br/>    reply = self.get_reply(client, expect_code = replies.RPL_YOURHOST, expect_nick = nick, expect_nparams = 1)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104b3b0b8&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104eb0f28&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="failed results-table-row">
        <tr>
          <td class="col-result">Failed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestQUIT::()::test_quit_after_registration4</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="log">self = &lt;chirc.tests.common.IRCSession object at 0x104ecc7b8&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104edeac8&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>&gt;           msg = client.get_message()<br/><br/>tests/chirc/tests/common.py:340: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.client.ChircClient object at 0x104edeac8&gt;<br/><br/>    def get_message(self):<br/>        msg = self.client.read_until(str.encode(&quot;\r\n&quot;), timeout=self.msg_timeout)<br/>        msg = msg.decode()<br/>        if msg[-2:] != &quot;\r\n&quot;:<br/>&gt;           raise ReplyTimeoutException(msg)<br/><span class="error">E           chirc.types.ReplyTimeoutException</span><br/><br/>tests/chirc/client.py:37: ReplyTimeoutException<br/><br/>During handling of the above exception, another exception occurred:<br/><br/>self = &lt;chirc.tests.test_connection.TestQUIT object at 0x104ecc710&gt;<br/>irc_session = &lt;chirc.tests.common.IRCSession object at 0x104ecc7b8&gt;<br/><br/>    def test_quit_after_registration4(self, irc_session):<br/>        &quot;&quot;&quot;<br/>            Connects two clients, and then has them quit, each with separate<br/>            messages.<br/>            &quot;&quot;&quot;<br/>    <br/>&gt;       client1 = irc_session.connect_user(&quot;user1&quot;, &quot;User One&quot;)<br/><br/>tests/chirc/tests/test_connection.py:512: <br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/>tests/chirc/tests/common.py:136: in connect_user<br/>    self.verify_welcome_messages(client, nick)<br/>tests/chirc/tests/common.py:444: in verify_welcome_messages<br/>    reply = self.get_reply(client, expect_code = replies.RPL_YOURHOST, expect_nick = nick, expect_nparams = 1)<br/>_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ <br/><br/>self = &lt;chirc.tests.common.IRCSession object at 0x104ecc7b8&gt;<br/>client = &lt;chirc.client.ChircClient object at 0x104edeac8&gt;, expect_code = &#x27;002&#x27;<br/>expect_nick = &#x27;user1&#x27;, expect_nparams = 1, expect_short_params = None<br/>long_param_re = None, long_param_values = None, expect_timeout = False<br/><br/>    def get_reply(self, client, expect_code = None, expect_nick = None, expect_nparams = None,<br/>                  expect_short_params = None, long_param_re = None, long_param_values = None,<br/>                  expect_timeout = False):<br/>        try:<br/>            msg = client.get_message()<br/>    <br/>            if expect_timeout:<br/>                pytest.fail(&quot;Was not expecting a reply, but got one:\n&quot; + msg.raw(bookends=True))<br/>        except EOFError:<br/>            pytest.fail(&quot;Server closed connection unexpectedly. Possible segfault in server?&quot;)<br/>        except ReplyTimeoutException as rte:<br/>            if expect_timeout:<br/>                return None<br/>    <br/>            if len(rte.bytes_received) == 0:<br/>                failmsg = &quot;Expected a reply but got none (no bytes received)&quot;<br/>            else:<br/>                failmsg = &quot;Expected a reply but did not get valid reply terminated with \\r\\n. Bytes received:\n|||{}|||&quot;.format(rte.bytes_received)<br/>&gt;           pytest.fail(failmsg)<br/><span class="error">E           Failed: Expected a reply but got none (no bytes received)</span><br/><br/>tests/chirc/tests/common.py:354: Failed<br/></div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_simple1</td>
          <td class="col-duration">0.18</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_simple2</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_both_messages_at_once</td>
          <td class="col-duration">0.00</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages1</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages2</td>
          <td class="col-duration">0.10</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages3</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages4</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages5</td>
          <td class="col-duration">0.21</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages6</td>
          <td class="col-duration">0.26</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_partitioned_messages7</td>
          <td class="col-duration">0.59</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_nick_user_parsing1</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_nick_user_parsing2</td>
          <td class="col-duration">0.22</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_nick_user_parsing3</td>
          <td class="col-duration">0.54</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_nick_user_parsing4</td>
          <td class="col-duration">0.27</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_nick_user_parsing5</td>
          <td class="col-duration">0.27</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome1</td>
          <td class="col-duration">0.10</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome2</td>
          <td class="col-duration">0.11</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome3</td>
          <td class="col-duration">0.22</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome4</td>
          <td class="col-duration">0.21</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome5</td>
          <td class="col-duration">0.21</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome6</td>
          <td class="col-duration">0.36</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody>
      <tbody class="passed results-table-row">
        <tr>
          <td class="col-result">Passed</td>
          <td class="col-name">tests/chirc/tests/test_connection.py::TestBasicConnection::()::test_connect_no_unexpected_welcome7</td>
          <td class="col-duration">0.38</td>
          <td class="col-links"></td></tr>
        <tr>
          <td class="extra" colspan="4">
            <div class="empty log">No log output captured.</div></td></tr></tbody></table></body></html>